tests/test_neoclassical.py

```diff
@@ -0,0 +1,170 @@
+    data = eq.compute("effective ripple", grid=grid)
+    assert np.isfinite(data["effective ripple"]).all()
+    eps_eff = grid.compress(data["effective ripple"])
+
+    neo_rho, neo_eps = NeoIO.read("tests/inputs/neo_out.w7x")
+    np.testing.assert_allclose(
+        eps_eff, np.interp(rho, neo_rho, neo_eps), rtol=0.16, atol=1e-5
+    )
+    fig, ax = plt.subplots()
+    ax.plot(rho, eps_eff, marker="o")
+    return fig
+
+
+class NeoIO:
+    """Class to interface with NEO."""
+
+    def __init__(self, name, eq, ns=256, M_booz=None, N_booz=None):
+        self.name = name
+        self.vmec_file = f"wout_{name}.nc"
+        self.booz_file = f"boozmn.{name}"
+        self.neo_in_file = f"neo_in.{name}"
+        self.neo_out_file = f"neo_out.{name}"
+
+        self.eq = eq
+        self.ns = ns  # number of surfaces
+        self.M_booz = setdefault(M_booz, 3 * eq.M + 1)
+        self.N_booz = setdefault(N_booz, 3 * eq.N)
+
+    @staticmethod
+    def read(name):
+        """Return ρ and ε¹ᐧ⁵ from NEO output with given name."""
+        with open(name) as f:
+            array = np.array([[float(x) for x in line.split()] for line in f])
+
+        neo_eps = array[:, 1]
+        neo_rho = np.sqrt(np.linspace(1 / (neo_eps.size + 1), 1, neo_eps.size))
+        # replace bad values with linear interpolation
+        good = np.isfinite(neo_eps)
+        neo_eps[~good] = np.interp(neo_rho[~good], neo_rho[good], neo_eps[good])
+        return neo_rho, neo_eps
+
+    def write(self):
+        """Write neo input file."""
+        self.eq.solved = True  # must set this for NEO to run correctly
+        print(f"Writing VMEC wout to {self.vmec_file}")
+        VMECIO.save(self.eq, self.vmec_file, surfs=self.ns, verbose=0)
+        self._write_booz()
+        self._write_neo()
+
+    def _write_booz(self):
+        print(f"Writing boozer output file to {self.booz_file}")
+        import booz_xform as bx
+
+        b = bx.Booz_xform()
+        b.read_wout(self.vmec_file)
+        b.mboz = self.M_booz
+        b.nboz = self.N_booz
+        b.run()
+        b.write_boozmn(self.booz_file)
+
+    def _write_neo(
+        self,
+        theta_n=200,
+        phi_n=200,
+        num_pitch=50,
+        multra=1,
+        acc_req=0.01,
+        nbins=100,
+        nstep_per=75,
+        nstep_min=500,
+        nstep_max=2000,
+        verbose=2,
+    ):
+        print(f"Writing NEO input file to {self.neo_in_file}")
+        f = open(self.neo_in_file, "w")
+
+        def writeln(s):
+            f.write(str(s))
+            f.write("\n")
+
+        # https://princetonuniversity.github.io/STELLOPT/NEO
+        writeln(f"'#' {datetime.now()}")
+        writeln(f"'#' {self.vmec_file}")
+        writeln(f"'#' M_booz={self.M_booz}. N_booz={self.N_booz}.")
+        writeln(self.booz_file)
+        writeln(self.neo_out_file)
+        writeln(self.ns - 1)
+        writeln(" ".join(str(i) for i in range(self.ns)))
```

Comment:
maybe @dpanici, should i start this from 1 instead of 0 for `neo_rho` as computed on line 90? these are the "surface indices", see above link

Reply:
I think it should be `range(2,self.ns+1)` actually, as all the neo_in files I see have that.

I think NEO does not compute it on-axis (so that is why `1` is excluded), and it starts indexing at `1` so `1` is the axis (corresponds to rho=0 or s=0 equivalently).

Reply:
Ok I did use ``range(2,self.ns+1)`` to generate the data and comparison plots, but upon refactoring this I wanted to check.